backend/dataall/modules/dataset_sharing/services/share_managers/lf_share_manager.py

```diff
@@ -63,6 +60,19 @@ def process_approved_shares(self) -> [str]:
     def process_revoked_shares(self) -> [str]:
         return NotImplementedError
 
+    def init_source_account_details(self):
+        """
+        Check if the catalog account is present and update the source account, source database and source region accordingly
+        """
```

Comment:
This function is a bit confusing, when you say "present" you mean that there is a separate catalog_account different from the source_environment.AwsAccount and that the catalog exists and is tagged right?